### PR TITLE
[Backbones] Update basic_training to support weight decay and ema

### PR DESCRIPTION
**Pull request by ianstenbit**
These are both Keras 2.11 features, so I've also updated the script to reference this dependency.

### CODE REVIEW

File: examples/training/classification/imagenet/basic_training.py [hunk 1]
```diff
@@ -74,6 +75,11 @@
     False,
     "Whether or not to use FP16 mixed precision for training.",
 )
+flags.DEFINE_boolean(
```
Comment (tanzhenyu) [inline examples/training/classification/imagenet/basic_training.py:78]: wanna also fix the random generator shape mismatch in this PR?

Comment (ianstenbit) [reply examples/training/classification/imagenet/basic_training.py:78]: Good call -- done

**Comment by tanzhenyu** [comment]
Doesn't look like code formatting is related to the PR. skipping the test
